app/src/org/commcare/activities/GeoPointMapActivity.java

```diff
@@ -153,7 +164,8 @@ private void loadProviders() {
     public void onLocationChanged(Location location) {
         if (!inViewMode && !isManualSelectedLocation) {
             if (location != null &&
-                    (this.location == null || (location.getAccuracy() < this.location.getAccuracy()))) {
```

Comment:
this was breaking when there is no initial location set on the map as new location's accuracy is always greater than 0 while base location's accuracy is set to 0 on start up

Reply:
Shouldn't the `this.location == null` check work when there's no initial location?

Reply:
location is [initialized](https://github.com/dimagi/commcare-android/blob/master/app/src/org/commcare/activities/GeoPointMapActivity.java#L54) to non null value to start with. So I don't think if it will ever be null.

Reply:
Got it, so `this.location == null` is never going to be true or do you think there can be a possibility of a null location?

Reply:
yeah, I don't think there is such a possibility though have not removed it to minimize changes in the hotfix